### ISSUE TITLE
Can't set multiple response cookies in handle hook

### ISSUE THREAD

**Issue by nbgoodall** [sveltejs/kit#4840]
### Describe the bug

Hello!

When setting multiple cookies from a hook like so:

```
export async function handle({ event, resolve }) {
  const response = await resolve(event);

  response.headers.set("set-cookie", ["a=b;", "b=c;", "c=d;"]);

  return response;
}
```

Only the first one, `a=b`, is set.

### Reproduction

https://github.com/nbgoodall/svelte-handle-not-setting-multiple-cookies

Run the above repo and either look on the homepage or inspect the browser cookies.

### System Info

```shell
System:
    OS: macOS 12.3.1
    CPU: (8) arm64 Apple M1
    Memory: 98.80 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.3.1 - /usr/local/bin/npm
  Browsers:
    Brave Browser: 101.1.38.109
    Safari: 15.4
    Safari Technology Preview: 15.4
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.40
    @sveltejs/kit: next => 1.0.0-next.326
    svelte: ^3.44.0 => 3.48.0
```

### Severity

blocking an upgrade

**Comment by tcc-sejohnson** [comment]
Am... I crazy? Maybe the behavior here was different in times past and I just never knew about it, but the signature of `Headers.set` [is (name, value)](https://developer.mozilla.org/en-US/docs/Web/API/Headers/set), where [value is, roughly speaking, a string](https://fetch.spec.whatwg.org/#header-value). I can't find anywhere that allows an array as the second argument to `Headers.set`, not even `node-fetch`. I know Kit allows headers returned from endpoints to be specified as an array, but that's a Kit thing -- the `Response` object returned from `handle` is just a `response`.

In fact, _everywhere I can find_ ([including MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie)) requires one `Set-Cookie` header per cookie value.

[The latest RFC I could find](https://www.rfc-editor.org/rfc/rfc6265) says:

> Origin servers SHOULD NOT fold multiple Set-Cookie header fields into a single header field.  The usual mechanism for folding HTTP headers fields (i.e., as defined in [[RFC2616](https://www.rfc-editor.org/rfc/rfc2616)]) might change the semantics of the Set-Cookie header field because the %x2C (",") character is used by Set-Cookie in a way that conflicts with such folding.

I can find no current endorsement of folding multiple `Set-Cookie` headers into one.

It's curious that `Headers.set` even allows a second argument as an array without failing, but it seems pretty clear from _all_ of the evidence I can find that a an array argument is not in the spec, and multiple `Set-Cookie` headers should be created by repeated calls to `Headers.append`.

TL;DR: I spent way too much time reading specs, and I don't think `Headers.set` is _supposed_ to take an array as its second argument. If ya wanna set multiple `set-cookie` headers, do:

```js
headers.set('set-cookie', 'name=val');
headers.set('set-cookie', 'name2=val2');
```

I'd love to find out I'm wrong, though, so please @ me if you find any current documentation contradicting this!

Typescript declaration of `Headers`: https://github.com/microsoft/TypeScript/blob/94391dc9b546e11ca2202557f3c8a299a0d3101a/lib/lib.dom.d.ts#L8874

Typescript declaration of `Response` (notice `headers` is just `Headers` as above):  https://github.com/microsoft/TypeScript/blob/94391dc9b546e11ca2202557f3c8a299a0d3101a/lib/lib.dom.d.ts#L12080

**Comment by nbgoodall** [comment]
Oh interesting! Appreciate the digging @tcc-sejohnson; I assumed that'd be the case given it worked ~100 versions ago and Kit's endpoint handling.

> If ya wanna set multiple set-cookie headers, do:

> ```
> headers.set('set-cookie', 'name=val');
> headers.set('set-cookie', 'name2=val2');
> ```

That should be append, right? The repeated `set` calls don't work (at least in Kit).

**Comment by dominikg** [comment]
Using `append` to add multiple set-cookie headers instead of `set` which would overwrite any existing values it works for me.

```js
export async function handle({ event, resolve }) {
  const response = await resolve(event);
  const cookies = {
    a: 'b',
    b: 'c',
    c: 'd',
  };
  Object.entries(cookies).forEach(e => response.headers.append('set-cookie', `${e.join('=')}`))
  console.log('headers', response.headers);
  return response;
}
```
logs the following on server console
```
headers {
 ...
  'set-cookie': [ 'a=b', 'b=c', 'c=d' ]
}
```
browser receives the headers
```
...
set-cookie: a=b
set-cookie: b=c
set-cookie: c=d
```

and has them available in document.cookie 
```
document.cookie
'a=b; b=c; c=d'
```

**Comment by tcc-sejohnson** [comment]
@nbgoodall, lol, yep, that should be `append`. I edited my original comment for future readers. 🙂

**Comment by tcc-sejohnson** [comment]
@benmccann / @Rich-Harris, sounds like this one can be closed UNLESS the maintainers want to try to override the behavior of `Headers.set` in `handle` to accommodate this use case.

**Comment by ITenthusiasm** [comment]
@benmccann Does this issue need a label? I feel like `wontfix` is a little misleading because it suggests that this is a bug when it isn't.

(I know this is a little random as a comment. I'm looking to make sure there are no duplicate issues before opening a new one and I stumbled across this one. Figured I'd comment on the label for historical purposes while I'm here.)
